Quartz/Engine/Include/Quartz/Core/Internals/QuartzDefines.hpp

```diff
@@ -23,43 +23,38 @@
 
 #pragma once
 
-#include <Quartz/Core/Core.hpp>
-
-#include <thread>
-#include <mutex>
-#include <condition_variable>
-
-#include <deque>
-#include <vector>
-#include <functional>
-
-namespace qz
-{
-	namespace utils
-	{
-		namespace threading
-		{
-			class ThreadPool
-			{
-			public:
-				ThreadPool(const int threadCount);
-				~ThreadPool();
-
-				void addWork(std::function<void()> fun);
-
-			private:
-				bool m_running;
-
-				std::mutex m_mutex;
-				std::condition_variable m_condition;
-
-				std::vector<std::thread> m_threads;
-				std::deque<std::function<void()>> m_scheduledTasks;
-
-			private:
-				void threadHandle();
-			};
-		}
-	}
-}
+#if defined(_DEBUG)
+#	define QZ_DEBUG
+#endif
+
+// Determine the compiler and set the corresponding defines for it.
+#ifdef _MSC_VER
+#	define QZ_MSVC
```

Comment:
define these as `QZ_COMPILER_*` instead of just `QZ_` to make it clear what they specify.